src/plugin.ts

```diff
@@ -21,11 +21,11 @@ export function buildEnd(err?: Error): void {
 }
 
 function useStyleSheetTransform(filename: string) {
-  return filename.match(/\.s?(a|c)ss/);
+  return path.extname(filename).match(/s?(a|c)ss/);
 }
 
 function useTypescriptTransform(filename: string) {
-  return filename.match(/\.tsx/);
+  return path.extname(filename).match('tsx');
```

Comment:
This change aligns the way that the file extensions are checked but is not crucial for fixing the bug. It could also perhaps be done as  `path.extname(filename) === 'tsx'` instead